### DISCUSSION TITLE
Clicking on the text field doesn't open the keyboard in Flutter application

### DISCUSSION THREAD

**Discussion by wdcs-princesoni** [appium/appium#20586] [Q&A]
Hello all,
Currently, we're facing an issue to enter the text in the text field for the Android application developed in Flutter. Developers added the identifier property as described [here](https://api.flutter.dev/flutter/semantics/SemanticsProperties/identifier.html) to get the element identified by resource-id Android. We're also able to identify the element using Appium inspector but when we select the element in Appium inspector then it highlights the entire page (Shown in the SS) instead of specific text field  to enter phone number. Because of that we we tries to set value using send_keys(), it just doesn't add the text but when we manually tap on that phone number field and hit the send_keys() method then it inserts the data in that field. 

Earlier we were using appium flutter driver but due to it's limitations and requirement of having the debug build in order to execute the test cases, we switched to Appium driver.

![Screenshot 2024-09-17 at 7 59 58 PM](https://github.com/user-attachments/assets/477c0561-eaae-42ef-a979-8711354c06fa)

Thank you in advance. 
@mykola-mokhnach your help will be much appreciated.

**Comment by mykola-mokhnach** [comment]
The inspector simply reflects what the app under test itself exposes to the accessibility. You need to change the app code in order to modify this behaviour.

**Comment by wdcs-princesoni** [comment]
This is the XML displaying in the Appium inspector.
               
               <android.widget.EditText index="0" package="com.zizle.dev" class="android.widget.EditText" text="" resource-id="phone_number_text_field" checkable="false" checked="false" clickable="true" enabled="true" focusable="true" focused="false" long-clickable="false" password="false" scrollable="false" selected="false" bounds="[0,414][1080,2273]" displayed="true" hint="Phone number&#10;You may receive SMS notifications from us for security and login purposes.">
                        <android.view.View index="0" package="com.zizle.dev" class="android.view.View" text="" content-desc="IN +91" resource-id="" checkable="false" checked="false" clickable="true" enabled="true" focusable="true" focused="false" long-clickable="false" password="false" scrollable="false" selected="false" bounds="[43,579][250,714]" displayed="true" />
                        <android.widget.Button index="1" package="com.zizle.dev" class="android.widget.Button" text="" content-desc="Continue" resource-id="" checkable="false" checked="false" clickable="false" enabled="false" focusable="true" focused="false" long-clickable="false" password="false" scrollable="false" selected="false" bounds="[43,937][1037,1072]" displayed="true" />
                     `
**This is the Flutter code of the Phone number text field where we added semantics identifier.** 👇 

    SizedBox(height: AppSize.h45),
                             Semantics(identifier: 'phone_number_text_field',child:  AppTextField(
                                key: const ValueKey('phone_number_text_field'),
                                hint: context.l10n.phoneNumber,
                                controller: signUpCubit.phoneNumberController,
                                keyboardType: TextInputType.phone,
                                inputFormatters: [
                                  FilteringTextInputFormatter.allow(RegExp('[0-9]')),
                                  LengthLimitingTextInputFormatter(12),
                                ],
                                `
 Can you suggest what changes we can make to identify the exact location/area of the element in Flutter code or anywhere else?
 
 Also, Just to find a workaround I've tried getting the relative coordinates of the phone number text field and tapping on it through the driver.tap() but that solution doesn't seem to be so effective as relative coordinates can vary device by device.
 
Any idea or suggestion would be really helpful.

**Comment by dlabs-matic-leva** [comment]
I'm commenting because this search result was on top of Google search for me. It doesn't solve your problem directly but I managed to put `content-desc` on Textfield which is close enough. My solution: https://stackoverflow.com/a/79609911/1274643

Copy of SO answer if link does down:

I couldn't confirm this with a published bug report but it seems that we can't set content-desc on TextField. I managed to find workaround by wrapping TextField in new Semantics widget and making it distinct SemanticsNode in semantics tree by setting container: true and explicitChildNodes: true. Code for reference:

```
Semantics(
  label: "Enter email address",
  container: true,
  explicitChildNodes: true,
  child: const TextField(),
);
```
This worked for Android and iOS, and Appium was able to select it on both platforms. This is how entering the text via Appium test looked like

```
await driver.$('~Enter email address').click();
await driver.sendKeys(process.env.TEST_USERNAME.split(''));
```
This also worked with FormBuilderTextField instead of TextField.
